ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -799,6 +811,22 @@ private boolean isInnerNode(final PlanNode node) {
     throw new IllegalStateException("Unknown node type: " + node.getClass().getName());
   }
 
+  private JoinKey buildForeignJoinKey(final Join join, final ColumnReferenceExp columnRef) {
```

Comment:
This method seems over complicated, and I am not even sure why we need it.

Seems a simple call to `JoinKey.foreignKeyColumn(leftSourceKeys);` should be sufficient? (or even push the extraction of the key columns into `foreignKeyColumn` and just pass in the "left input"?